examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,420 @@
+# This particular dataset has a stepwise constant target for the first 17,760
+# samples:
+
+df["transfer"][:17_760].unique()
+
+# %%
+# Let us drop those entries and explore the hourly electricity transfer over
+# different days of the week:
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+df = electricity.frame.iloc[17_760:]
+X = df.drop(columns=["transfer", "class"])
+y = df["transfer"]
+
+fig, ax = plt.subplots(figsize=(15, 10))
+pointplot = sns.lineplot(x=df["period"], y=df["transfer"], hue=df["day"], ax=ax)
+handles, lables = ax.get_legend_handles_labels()
+ax.set(
+    title="Hourly energy transfer for different days of the week",
+    xlabel="Normalized time of the day",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend(handles, ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"])
+
+# %%
+# Notice energy transfer increases systematically during weekends.
+#
+# Effect of number of trees and early stopping
+# ============================================
+# For the sake of illustrating the effect of the (maximum) number of trees, we
+# train a :class:`~sklearn.ensemble.HistGradientBoostingRegressor` over the
+# daily electricity transfer using the whole dataset. Then we visualize its
+# predictions depending on the `max_iter` parameter. Here we don't try to
+# evaluate the performance of the model and its capacity to generalize but
+# rather its capacity to learn from the training data.
+
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4, shuffle=False)
+
+print(f"Training sample size: {X_train.shape[0]}")
+print(f"Test sample size: {X_test.shape[0]}")
+print(f"Number of features: {X_train.shape[1]}")
+
+# %%
+max_iter_list = [5, 50]
+average_week_demand = (
+    df.loc[X_test.index].groupby(["day", "period"], observed=False)["transfer"].mean()
+)
+colors = sns.color_palette("colorblind")
+fig, ax = plt.subplots(figsize=(10, 5))
+average_week_demand.plot(color=colors[0], label="recorded average", linewidth=2, ax=ax)
+
+for idx, max_iter in enumerate(max_iter_list):
+    hgbt = HistGradientBoostingRegressor(max_iter=max_iter, categorical_features=None)
+    hgbt.fit(X_train, y_train)
+
+    y_pred = hgbt.predict(X_test)
+    prediction_df = df.loc[X_test.index].copy()
+    prediction_df["y_pred"] = y_pred
+    average_pred = prediction_df.groupby(["day", "period"], observed=False)[
+        "y_pred"
+    ].mean()
+    average_pred.plot(
+        color=colors[idx + 1], label=f"max_iter={max_iter}", linewidth=2, ax=ax
+    )
+
+ax.set(
+    title="Predicted average energy transfer during the week",
+    xticks=[(i + 0.2) * 48 for i in range(7)],
+    xticklabels=["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend()
+
+# %%
+# With just a few iterations, HGBT models can achieve convergence (see
```

Comment:
I think we need a sentence to describe what we see on the figure. 5 iterations is not enough to be able to predict. But at 50, we are already able to do a good job.

Reply:
> The variable `day` is actually a "category" variable.

The thing is that `day` is already ordinal encoded, so I don't think we can demo the categorical variables support here.

Reply:
It does not really matter. Since this is a category columns, the tree will treat it as such.

A potential thing that we could do is to replace the integers by the day as string at the beginning. I don't know if eventually it removes some boilerplate when plotting because we already have the name instead of remapping the integer to the days.

Reply:
This said, I would be happy to merge this PR as-is because it is a net improvement and see if we can further improve it with this increment.